ocaml/sdk-gen/go/gen_go_binding.ml

```diff
@@ -44,20 +44,84 @@ let render_api_messages_and_errors destdir =
   generate_file ~rendered:messages_rendered ~destdir
     ~output_file:"api_messages.go"
 
+let render_convert_header () =
+  let obj =
+    `O
+      [
+        ("name", `String "convert")
+      ; ( "modules"
+        , `O
+            [
+              ("import", `Bool true)
+            ; ( "items"
+              , `A
+                  [
+                    `O [("name", `String "fmt"); ("sname", `Null)]
+                  ; `O [("name", `String "math"); ("sname", `Null)]
+                  ; `O [("name", `String "reflect"); ("sname", `Null)]
+                  ; `O [("name", `String "strconv"); ("sname", `Null)]
+                  ; `O [("name", `String "time"); ("sname", `Null)]
+                  ]
+              )
+            ]
+        )
+      ]
+  in
+  render_template "FileHeader.mustache" obj ~newline:true ()
+
+let render_converts converts destdir =
+  let templates =
+    [
+      "ConvertSimpleType.mustache"
+    ; "ConvertInt.mustache"
+    ; "ConvertFloat.mustache"
+    ; "ConvertTime.mustache"
+    ; "ConvertRef.mustache"
+    ; "ConvertSet.mustache"
+    ; "ConvertRecord.mustache"
+    ; "ConvertInterface.mustache"
+    ; "ConvertMap.mustache"
+    ; "ConvertEnum.mustache"
+    ; "ConvertBatch.mustache"
+    ; "ConvertOption.mustache"
+    ]
+  in
+  let rendered =
+    let rendered =
+      templates
+      |> List.map (fun template -> render_template template converts ())
+      |> String.concat ""
+      |> String.trim
+    in
+    render_convert_header () ^ rendered ^ "\n"
+  in
+  generate_file ~rendered ~destdir ~output_file:"convert.go"
+
 let main destdir =
   render_api_messages_and_errors destdir ;
   let enums = Json.all_enums objects in
   render_enums enums destdir ;
-  let objects = Json.xenapi objects in
+  let objects, converts = objs_and_convert_functions objects in
```

Comment:
I think this `objs_and_convert_functions` does two un-related things. Could it be split?

Reply:
Yeah, it could be split and it's the right way.

Reply:
These related code has been split to the following PR which will raised after this one got merged.